test/copy.jl

```diff
@@ -253,6 +253,14 @@ end
     @test copyto!(s, String[]) == [1, 2] # No error
 end
 
+@testset "circular reference arrays" begin
+    # issue 56775
+    p = Any[nothing]
+    p[1] = p
+    p2 = deepcopy(p)
+    @test p2 === p2[1]
```

Comment:
Wouldn't hurt, I guess:

```suggestion
    @test p2 === p2[1]
    @test p2 !== p
```